routes/user.js

```diff
@@ -2,100 +2,123 @@
 
 const AppConstants = require("../app-constants");
 
-const crypto = require("crypto");
 const express = require("express");
 const router = express.Router();
 
 const EmailUtils = require("../email-utils");
 const Subscribers = require("../subscribers");
 
-// We send verification emails to addresses that want to subscribe.
-// These addresses are temporarily stored here, mapped to the unique
-// string token that is used for verification.
-const gUnverifiedEmails = new Map();
+const ResponseCodes = Object.freeze({
+  InternalError: 999,
+  EmailNotProvided: 100,
+  EmailNotFound: 101,
+  TokenMismatch: 102,
+});
+
+function resDatabaseError(res) {
+  console.log(error);
+  res.status(500).json({
+    code: 500,
```

Comment:
I'll argue that returning `code` in the JSON response may be all that useful since it just seems to echo the server response code. Plus, using `fetch()` or `XMLHttpRequest()` or `request()`/`got()` should give the user that information if they need it. And having `code` and `error_code` in the response may be confusing (not that I expect people to hit this API directly, I guess -- so maybe all of this is acedemic).

@groovecoder Thoughts?

Reply:
Hmm, I copied this from the fxa-oauth-server API [1]
I figured it's just for consumers' convenience, I don't care about it, I'm okay with not including the HTTP code.

[1] https://github.com/mozilla/fxa-oauth-server/blob/master/docs/api.md